### PR TITLE
Upload Image

### LINKED ISSUE #69: Make the upload button actually do something
When creating a post and the user uploads an image, we want to show a preview in the create post dialog.

### PR DESCRIPTION
**Pull request by maharshmellow**
Fixes #69 

Features:
- drag and drop to upload image
- show a preview at the bottom of the modal before adding the image to the post
- shows a preview of the uploaded image + text on the preview modal
- allows the user to choose another image after it has already been uploaded
- if more than one file is dragged, it will give an error
- if a non-image file is dragged, it will give an error

### CODE REVIEW

File: src/styles/post/UploadImageModal.scss [hunk 1]
```diff
@@ -2,6 +2,29 @@
...
 .upload-image-modal {
   @extend %modal;
+
+  .dropzone {
+    text-align: center;
+    padding: 20px;
+    border-width: 2px;
+    border-radius: 5px;
+    border-style: dashed;
+    background-color: #fafafa;
```
Comment (ceciliaccwei) [inline src/styles/post/UploadImageModal.scss]: since the background-color is already in the common stylesheet, can we just use that variable directly?

**Review by ceciliaccwei — CHANGES_REQUESTED** [review]
I checked it out locally and went over the code. Everything looks good except for that one color css😄

File: src/styles/post/UploadImageModal.scss (hunk 1 above)
Comment (maharshmellow) [reply src/styles/post/UploadImageModal.scss]: Done!

**Review by ceciliaccwei — APPROVED** [review]
looks good😄

**Pull request merged**